api/app/jobs/common_model_fetchers.py

```diff
@@ -353,9 +354,24 @@ def _process_model_run_for_station(self,
             query = get_model_run_predictions_for_grid(
                 self.session, model_run, grid)
 
+            cumulative_precip = [0.0, 0.0, 0.0, 0.0]
             # Iterate through all the predictions.
             prev_prediction = None
+
+            # 00 and 12 hour model runs accumulate precipitation in 12 hour intervals, 06 and 18 hour accumlate in
```

Comment:
Makes sense, just wondering if we want to add some tests since due to the complexity.

Reply:
Hmmm...I'll see if I can factor all this code out into a utility function that we can test. Unless you have another thought for how we might test?